ext/rotoscope/rotoscope.c

```diff
@@ -156,12 +158,15 @@ static rs_tracepoint_t extract_full_tracevals(rb_trace_arg_t *trace_arg, const r
   rs_class_desc_t method_owner = tracearg_class(trace_arg);
   rb_event_flag_t event_flag = rb_tracearg_event_flag(trace_arg);
 
+  VALUE method_name = tracearg_method_name(trace_arg);
+  VALUE filepath = callsite->filepath;
+
   return (rs_tracepoint_t){
       .event = evflag2name(event_flag),
-      .entity = method_owner.name,
-      .method_name = tracearg_method_name(trace_arg),
+      .entity = StringValueCStr(method_owner.name),
+      .method_name = StringValueCStr(method_name),
       .method_level = method_owner.method_level,
-      .filepath = callsite->filepath,
+      .filepath = StringValueCStr(filepath),
       .lineno = callsite->lineno};
```

Comment:
Hmm good point, but I think these both have the same issue in that both the `VALUE` and `char *` could be GC'd at any time.

> The lifetime of the C memory referenced by the result of RSTRING_PTR [used by StringValueCStr] is the same as the garbage collection lifetime of the corresponding String object.

I think the only way around this is to copy the strings over to some manually-allocated memory.